client/utils/auth.py

```diff
@@ -104,17 +109,19 @@ def update_storage(access_token, expires_in, refresh_token):
             'refresh_token': refresh_token
         }, fp)
 
-def authenticate(force=False):
-    """
-    Returns an oauth token that can be passed to the server for identification.
+def authenticate(args, force=False):
+    """Returns an OAuth token that can be passed to the server for
+    identification. If FORCE is False, it will attempt to use a cached token
+    or refresh the OAuth token. ARGS is the command-line arguments object.
     """
+    server = '{}://{}'.format('http' if args.insecure else 'https', args.server)
```

Comment:
Thought about it some more and I'm okay with it. Once we deploy this and if it seems to work - we can make it so that it only allows `http` when the server is `localhost`.